src/Swarm/Docs/Pedagogy.hs

```diff
@@ -0,0 +1,201 @@
+
+-- |
+-- SPDX-License-Identifier: BSD-3-Clause
+--
+-- Assess pedagogical soundness of the tutorials.
+--
+-- Approach:
+-- 1. Obtain a list of all of the tutorial scenarios, in order
+-- 2. Search their "solution" code for `commands`
+-- 3. "fold" over the tutorial list, noting which tutorial was first to introduce each command
+module Swarm.Docs.Pedagogy (
+  renderTutorialProgression,
+  generateIntroductionsSequence,
+  CoverageInfo (..),
+  TutorialInfo (..),
+) where
+
+import Control.Arrow ((&&&))
+import Control.Lens (universe, view)
+import Control.Monad (guard)
+import Control.Monad.Except (ExceptT (..), liftIO)
+import Data.Char (isLetter)
+import Data.List (foldl', sort)
+import Data.List.Split (wordsBy)
+import Data.Map qualified as M
+import Data.Maybe (mapMaybe)
+import Data.Set (Set)
+import Data.Set qualified as S
+import Data.Text (Text)
+import Data.Text qualified as T
+import Swarm.Docs.Util
+import Swarm.Game.Entity (loadEntities)
+import Swarm.Game.Scenario (Scenario, scenarioDescription, scenarioName, scenarioObjectives, scenarioSolution)
+import Swarm.Game.Scenario.Objective (objectiveGoal)
+import Swarm.Game.ScenarioInfo (ScenarioCollection, ScenarioInfoPair, flatten, loadScenariosWithWarnings, scenarioCollectionToList, scenarioPath)
+import Swarm.Language.Module (Module (..))
+import Swarm.Language.Pipeline (ProcessedTerm (..))
+import Swarm.Language.Syntax
+import Swarm.Language.Types (Polytype)
+import Swarm.TUI.Controller (getTutorials)
+import Swarm.Util (commaList)
+
+-- * Constants
+
+wikiPrefix :: Text
+wikiPrefix = "https://github.com/swarm-game/swarm/wiki/"
+
+commandsWikiPrefix :: Text
+commandsWikiPrefix = wikiPrefix <> "Commands-Cheat-Sheet#"
+
+-- * Types
+
+-- | Tutorials augmented by the set of
+-- commands that they introduce.
+-- Generated by folding over all of the
+-- tutorials in sequence.
+data CoverageInfo = CoverageInfo
+  { tutInfo :: TutorialInfo
+  , novelSolutionCommands :: Set Const
+  }
+
+-- | Tutorial scenarios with the set of commands
+-- introduced in their solution and descriptions
+-- having been extracted
+data TutorialInfo = TutorialInfo
+  { scenarioPair :: ScenarioInfoPair
+  , solutionCommands :: Set Const
+  , descriptionCommands :: Set Const
+  }
+
+-- | A private type used by the fold
+data CommandAccum = CommandAccum
+  { _encounteredCmds :: Set Const
+  , tuts :: [CoverageInfo]
+  }
+
+-- * Functions
+
+-- | Extract commands from both goal descriptions and solution code.
+extractCommandUsages :: ScenarioInfoPair -> TutorialInfo
+extractCommandUsages siPair@(s, _si) =
+  TutorialInfo siPair solnCommands $ getDescCommands s
+ where
+  solnCommands = S.fromList $ maybe [] getCommands maybeSoln
+  maybeSoln = view scenarioSolution s
+
+-- | Obtain the set of all commands mentioned by
+-- name in the tutorial's goal descriptions.
+--
+-- NOTE: It may be more robust to require that a command reference
+-- be surrounded by backticks and parse for that accordingly.
```

Comment:
Yes, I was going to suggest this.  Based on the sample output in the gist you linked, it seems like there are many false positives where we happened to use an English word which is also the name of a command.